internal/extproc/messages_processor.go

```diff
@@ -273,6 +275,14 @@ func (c *messagesProcessorUpstreamFilter) ProcessResponseBody(ctx context.Contex
 			c.metrics.RecordRequestCompletion(ctx, true, c.requestHeaders)
 		}
 	}()
+	if code, _ := strconv.Atoi(c.responseHeaders[":status"]); !isGoodStatusCode(code) {
+		// For now, simply pass through error responses without modification.
+		// TODO: do the error conversion like other processors, to be able to capture any error in the proper
```

Comment:
this is a serious TODO, which basically means that this processor has been very fragile until we solve this